src/index.js

```diff
@@ -73,13 +74,39 @@ function Comp3() {
   </div>)
 }
 
-function App() {
+class Comp4 extends React.Component {
+  render() {
+    const {count} = this.props;
+    console.log('render HOC');
+    return (
+      <div>
+        withCounter.count = {count}<br/>
+      </div>
+    );
+  }
+}
 
+const WrappedComp4 = withReuse(counter, ({count}) => ({count}))(Comp4);
```

Comment:
I'm not sure that's a good choice for mapStateToProps syntax. You'll usually want to bind a few parameters from state and this way you'll have to create several of these for each component.

What about the following syntax instead which will allow multiple parameters:
```jsx harmony
const WrappedComp4 = withReuse([counter, modulo], ([counter, modulo] => ({ count: counter.count, modulo: modulo })(Comp4)
```

Or maybe
```jsx harmony
const WrappedComp4 = withReuse([
  { unit: counter, getValue: ({ count}) => count] },
  { unit: modulo, getValue: 'modulo' } // alternative string based syntax 
)(Comp4)
```